### PR TITLE
changes to compile on AWS

### PR DESCRIPTION
**Pull request by weihuang-jedi**
Add a modulefile to be used when compile on AWS.

This PR is related to issue: 
https://github.com/NOAA-EMC/gfs-utils/issues/65

### CODE REVIEW

File: modulefiles/gfsutils_noaacloud.intel.lua [hunk 1]
```diff
@@ -0,0 +1,17 @@
+help([[
+Build environment for GFS utilities on Hera
```
Comment (aerorahul) [inline modulefiles/gfsutils_noaacloud.intel.lua]:
```suggestion
Build environment for GFS utilities on NOAA cloud
```

File: modulefiles/gfsutils_noaacloud.intel.lua [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+local stack_impi_ver=os.getenv("stack_impi_ver") or "2021.3.0"
+local cmake_ver=os.getenv("cmake_ver") or "3.20.1"
+
+load(pathJoin("stack-intel", stack_intel_ver))
+load(pathJoin("stack-intel-oneapi-mpi", stack_impi_ver))
+load(pathJoin("cmake", cmake_ver))
+
+load("gfsutils_common")
+
+whatis("Description: GFS utilities environment on Hera with Intel Compilers")
```
Comment (aerorahul) [inline modulefiles/gfsutils_noaacloud.intel.lua]:
```suggestion
whatis("Description: GFS utilities environment on NOAA cloud with Intel Compilers")
```

File: modulefiles/gfsutils_common.lua [hunk 3]
```diff
@@ -5,7 +5,7 @@ Load common modules to build GFS utilities on all machines
 local netcdf_c_ver=os.getenv("netcdf_c_ver") or "4.9.2"
 local netcdf_fortran_ver=os.getenv("netcdf_fortran_ver") or "4.6.1"
 
-local bufr_ver=os.getenv("bufr_ver") or "11.7.0"
+local bufr_ver=os.getenv("bufr_ver") or "12.0.1"
```
Comment (aerorahul) [inline modulefiles/gfsutils_common.lua]: Will this work on on-prem platforms if 12.0.1 is not available in spack-stack 1.6.0 unified env

**Review by aerorahul — CHANGES_REQUESTED** [review]
some minor comments on the noaa-cloud modulefile.
However, changing of the default for bufr to 12.0 will likely break this system.
Can we have 11.7.0 installed on the cloud contrib/ space?  It will keep the cloud in sync w/ on prem and ease future upgrades.

File: modulefiles/gfsutils_common.lua (hunk 3 above)
Comment (weihuang-jedi) [reply modulefiles/gfsutils_common.lua]: I'll do check to see if bufr 11.7.0 can be loaded on AWS.

**Review by aerorahul — APPROVED** [review]
looks good.

**Pull request merged**